### PR TITLE
Add new methods

### LINKED ISSUE #179: Add get_class_by_hash method

### LINKED ISSUE #180: Add get_class_hash_at method

### PR DESCRIPTION
**Pull request by cptartur**
### **Please check if the PR fulfills these requirements**
- [ ] The formatter, linter, and tests all run without an error
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

## **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* Adds get_class_hash_at (Closes #180)
* Adds get_class_by_hash (Closes #179)

 **What is the current behavior?** (You can also link to an open issue here)

n/a

## **What is the new behavior (if this is a feature change)?**

Added missing methods

## **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

No

## **Other information**

n/a

### CODE REVIEW

File: starknet_py/net/client.py [hunk 1]
```diff
@@ -261,3 +264,21 @@ async def deploy(
             raise Exception("Transaction not received")
 
         return res
+
+    async def get_class_hash_at(
+        self,
+        contract_address: CastableToHash,
+        block_hash: Optional[CastableToHash] = None,
+        block_number: Optional[BlockIdentifier] = None,
+    ) -> str:
```
Comment (Solpatium) [inline starknet_py/net/client.py:273]: Shoudldn't we return an int?

**Review by Solpatium — APPROVED** [review]

File: starknet_py/net/client.py (hunk 1 above)
Comment (cptartur) [reply starknet_py/net/client.py:273]: Gateway's get_class_hash_at returns string, we could convert it to int if that's better.

**Review by Arcticae — CHANGES_REQUESTED** [review]

File: starknet_py/net/client.py [hunk 2]
```diff
@@ -265,3 +265,35 @@ async def deploy(
...
+
+    async def get_class_by_hash(self, class_hash: CastableToHash) -> Dict[str, Any]:
+        """
+        Retuns the contract class for given hash
+
+        :param class_hash: Class hash
+        :return: Dict with representation of contract class
+        """
+        if isinstance(class_hash, int):
+            class_hash = hex(class_hash)
```
[the remark this answers is not in the document]
Comment (cptartur) [reply starknet_py/net/client.py:297]: Yes, I was waiting for devnet support for declare transactions. Going to add some test later.

File: starknet_py/tests/e2e/client/client_test.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+async def test_get_class_hash_at(run_devnet):
+    client = await DevnetClientFactory(run_devnet).make_devnet_client()
+
+    deployment_result = await Contract.deploy(
+        compilation_source=map_source, client=client
+    )
+    await deployment_result.wait_for_acceptance()
+    contract = deployment_result.deployed_contract
+
+    await client.get_class_hash_at(contract_address=contract.address)
```
[the remark this answers is not in the document]
Comment (cptartur) [reply starknet_py/tests/e2e/client/client_test.py:24]: I'll try to improve that

File: starknet_py/net/client.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Arcticae) [reply starknet_py/net/client.py:343]: This seems like it's untested still

File: starknet_py/tests/e2e/client/client_test.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+
+@pytest.mark.asyncio
+async def test_get_class_by_hash(run_devnet):
+    client = await DevnetClientFactory(run_devnet).make_devnet_client()
+
+    declare_result = await client.declare(compilation_source=map_source)
+    class_hash = declare_result["class_hash"]
+
+    assert await client.get_class_by_hash(class_hash) is not None
```
Comment (Arcticae) [inline starknet_py/tests/e2e/client/client_test.py:47]: same here

**Review by Arcticae — CHANGES_REQUESTED** [review]

**Pull request merged**